From 72f3edd9131db2bbd31a293595fb25bea5cb1237 Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Mon, 26 Feb 2024 23:01:58 +0530
Subject: [PATCH 0913/1141] dt-bindings: phy: Add wakeup-source for STM32MP25
 COMBOPHY bindings

Add wakeup-source capability

Change-Id: I79894b829727fea2cb3422824e478755a42cad99
Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/363203
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../bindings/phy/st,stm32-combophy.yaml       | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

--- a/Documentation/devicetree/bindings/phy/st,stm32-combophy.yaml
+++ b/Documentation/devicetree/bindings/phy/st,stm32-combophy.yaml
@@ -87,6 +87,12 @@ properties:
       Voltage Swing to change the Vlo, Vhi for a VDDP = 0.8V.
       The hardware accepts discrete values in this range.
 
+  wakeup-source: true
+
+  interrupts:
+    maxItems: 1
+    description: interrupt used for wakeup
+
 required:
   - compatible
   - reg
@@ -97,6 +103,15 @@ required:
   - clocks
   - clock-names
 
+allOf:
+  - if:
+      required:
+        - wakeup-source
+    then:
+      anyOf:
+        - required: [interrupts]
+        - required: [interrupts-extended]
+
 additionalProperties: false
 
 examples:
@@ -114,6 +129,8 @@ examples:
               resets = <&rcc USB3PCIEPHY_R>;
               reset-names = "phy-rst";
               st,syscfg = <&syscfg>;
+              wakeup-source;
+              interrupts-extended = <&exti1 45 IRQ_TYPE_EDGE_FALLING>;
     };
 
   - |
@@ -139,5 +156,7 @@ examples:
               resets = <&rcc USB3PCIEPHY_R>;
               reset-names = "phy-rst";
               st,syscfg = <&syscfg>;
+              wakeup-source;
+              interrupts-extended = <&exti1 45 IRQ_TYPE_EDGE_FALLING>;
     };
 ...
